### PR TITLE
feat: added way to include files to application package

### PR DESCRIPTION
**Pull request by BrageHK**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by BrageHK** [comment]
@glebashnik Just let me know I should ask someone else for review or if it would be better to have a .vespaignore pattern instead of explicitly including files.

**Comment by glebashnik** [comment]
I will take a look today.

**Comment by glebashnik** [comment]
Check out copilot feedback.

**Review by glebashnik — CHANGES_REQUESTED** [review]
1. Consider specifying source and target file paths explicitely, e.g. include 
models/embedders/qwen_tuned_2026-01-07.onnx as model/qwen_embedder.onnx in application package.
We can use tuples, dictionary or dataclass to specify source and target file.
 
2. Perhaps we should handle files colliding with application files such as services.xml better.

3. Tests lack edge cases like file not found, absolute path, specifying directory instead of file, collision with services.xml.

File: vespa/package.py [hunk 1]
```diff
@@ -3565,6 +3577,13 @@ def to_files(self, root: Path) -> None:
             with open(os.path.join(root, "deployment.xml"), "w") as f:
                 f.write(self.deployment_to_text)
 
+        for path in self.include_files:
+            p = Path(path)
+            dest_rel = str(path) if not p.is_absolute() else p.name
```
Comment (glebashnik) [inline vespa/package.py]: File checks and sanitization logic appears in several places, can be a separate function/method.

**Review by glebashnik — CHANGES_REQUESTED** [review]

**Comment by BrageHK** [comment]
Everything should be fixed now. Most of the code was made with AI.

File: vespa/package.py [hunk 2]
```diff
@@ -3092,7 +3092,42 @@ def validate(self):
...
+
+        Raises ``ValueError`` if the path is absolute or contains any ``..`` component.
+        ``..`` is rejected even when embedded (e.g. ``a/../b``), so the caller always
+        gets a clean, forward-only relative path.
+        """
+        p = Path(dest)
+        # On Windows, Path("/foo").is_absolute() is False (drive-relative), so also
+        # check for a leading "/" to catch POSIX-style absolute paths on all platforms.
+        if p.is_absolute() or str(dest).startswith("/"):
+            raise ValueError(
```
[the remark this answers is not in the document]
Comment (glebashnik) [reply vespa/package.py:3117]: Minor, non-blocking.

**Review by glebashnik — APPROVED** [review]
